pkg/api/read.go

```diff
@@ -88,7 +88,7 @@ func Read(config *Config, reader querier.Reader, metrics *Metrics, updateMetrics
 		compressed = snappy.Encode(nil, data)
 		if _, err := w.Write(compressed); err != nil {
 			statusCode = "500"
-			http.Error(w, err.Error(), http.StatusInternalServerError)
+			log.Warn("msg", "Error writing HTTP response", "err", err)
```

Comment:
This should be logged as error i think.

Reply:
And do we respond back in HTTP in an error case?

Reply:
In keeping with previously made assumptions, using Warn here as there is a similar situation above:

```log.Warn("msg", "Error executing query", "query", req, "storage", "PostgreSQL", "err", err)```

Not sure about your HTTP comment. Client aborted the request, whatever we do with the HTTP response its not going to reach them.